src/c76_menuCAL.hpp

```diff
@@ -1231,28 +1248,46 @@ void printCalibrationSubmenu()
 #if USE_GYRO_LEVEL == 1
   else if (calState == ROLL_OFFSET_CALIBRATION)
   {
+    calCount++;
     auto angles = Gyro::getCurrentAngles();
-    sprintf(scratchBuffer, "R: -------------");
-    makeIndicator(scratchBuffer, angles.rollAngle - mount.getRollCalibrationAngle());
-    lcdMenu.printMenu(scratchBuffer);
+    accumulator.rollAngle += angles.rollAngle;
+    if (calCount >= ACCUMULATOR_COUNT)
+    {
+      accumulator.rollAngle /= (float)ACCUMULATOR_COUNT;
```

Comment:
Could you instead implement the details in the gyro module instead of the menu file? It doesn't make sense to do this here.

Also while we're changing this, doing this dance with a calibration counter, accumulator, reset, etc seems like a very simplistic way to implement a low pass filter. Something like an exponential low pass filter ([example C implementation](https://kiritchatterjee.wordpress.com/2014/11/10/a-simple-digital-low-pass-filter-in-c/)) is very simple to implement and would work fantastic for our use case. :smile: 
I can definitely help you out here if you need it!

Reply:
I don't want the Gyro class to do this, it would make it unneccesarily slower. But here the frequent LCD updates make it unreadable. I want it a lot slower, but only for the LCD. So this is the right place. Won't fix.

Reply:
OK, that low pass filter works really well (in the test app). Trying to see if I can get it to behave in the firmware